File: web/backend/database/models/Parent.js
const mongoose = require('mongoose');
const { COLLECTIONS } = require('../constants');

const ParentSchema = new mongoose.Schema({
  parentId: {
    type: Number,
    required: true,
    unique: true
  },
  userId: {
    type: String,
    required: true,
    unique: true
  },
  fullName: {
    type: String,
    required: true
  },
  career: {
    type: String
  },
  phone: {
    type: String
  },
  gender: {
    type: Boolean
  },
  studentIds: [{
    type: Number,
    ref: 'Student'
  }]
}, {
  toJSON: { virtuals: true },
  toObject: { virtuals: true }
});

// Virtual for getting user info
ParentSchema.virtual('user', {
  ref: 'User',
  localField: 'userId',
  foreignField: 'userId',
  justOne: true
});

// Virtual for getting students
ParentSchema.virtual('students', {
  ref: 'Student',
  localField: 'studentIds',
  foreignField: 'studentId',
  justOne: false
});

module.exports = mongoose.model('Parent', ParentSchema, COLLECTIONS.PARENT); 

File: web/backend/Python-services/src/models/Parent.py
"""
Parent model for FAMS
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field
from ..constants import COLLECTIONS


class Parent(BaseModel):
    """Parent model representing student guardians"""
    parentId: int = Field(..., description="Unique identifier for the parent")
    userId: int = Field(..., description="Reference to UserAccount")
    fullName: str = Field(..., description="Parent's full name")
    career: Optional[str] = Field(None, description="Parent's career/occupation")
    phone: Optional[str] = Field(None, description="Parent's phone number")
    gender: bool = Field(..., description="Parent's gender (True for Male, False for Female)")
    createdAt: datetime = Field(default_factory=datetime.utcnow)
    updatedAt: datetime = Field(default_factory=datetime.utcnow)
    isActive: bool = Field(default=True)
    collection: str = COLLECTIONS['PARENT'] 